### PR TITLE
feat: added patreon oidc provider

### PR DESCRIPTION
**Pull request by AUSBird**
This PR is intending to implement another OIDC provider for [Patreon](https://docs.patreon.com/#apiv2-oauth) to Kratos since Patreon has non-standard paths to it's `auth` and `token` endpoints and doesn't have a discovery endpoint

While I was there I also added missing OIDC provider types to a code comment to update it with missing ones (Not sure if it matters or no)

## Related issue(s)

I never made an issue to for this because it's just a new provider and the issue templates only has one for non-trivial features. (If I should have made one anyway, tell me for next time :) )

I did bring it up months ago in slack but never got any response, 

PR for Documentation: https://github.com/ory/docs/pull/1211

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [ ] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got the approval (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added or changed [the documentation](https://github.com/ory/docs).

## Further Comments

This is my first ever *go* at golang so feel free to give me feedback on anything but I do ask you simplify a bit and if possible can I get pointers to documentation when comments are made to help me learn :)

### CODE REVIEW

**Comment by AUSBird** [comment]
I don't know how or where that unit test broke, I had a look where it said and nothing around it seems to obviously connect to my changes.

Can someone please give me a small pointer to solving it?

File: selfservice/strategy/oidc/provider_patreon.go
Comment (zepatrik) [inline selfservice/strategy/oidc/provider_patreon.go:83]: Please use the resilient client, similar to https://github.com/ory/kratos/blob/f2960124cf62886025b775a06c85e697270f8d6a/selfservice/strategy/oidc/provider_auth0.go#L79
It uses some retry logic. Also, you should handle the error here (although it should never error) to prevent a later update to break this.

**Review by zepatrik — CHANGES_REQUESTED** [review]
Nice, this looks very good :+1:
I just have one minor remark.

File: selfservice/strategy/oidc/provider_patreon.go
Comment (AUSBird) [reply selfservice/strategy/oidc/provider_patreon.go]: I made the requested change :)

Do the maintainers want me to resolve conversations when I make the change or do the maintainers want me to leave them open for you to change when you have reviewed the change?

Comment (jonas-jonas) [reply selfservice/strategy/oidc/provider_patreon.go]: Typically, the creator of the comment will resolve it themselves, after whether the comment is resolved. :)

**Review by jonas-jonas — APPROVED** [review]
Looks very good to me. Thanks for your contribution. :)

Waiting for @aeneasr or @zepatrik to confirm & merge.

**Review by zepatrik — APPROVED** [review]
LGTM :+1:

**Comment by vinckr** [comment]
Hello @AUSBird 
Congrats on merging your first PR in Ory 🎉 !
Your contribution will soon be helping secure millions of identities around the globe 🌏. 
As a small token of appreciation we send our first time contributors a gift package to welcome them to the community. 
[Please drop me an email and I will forward you the form to claim your Ory swag!](mailto:vincent@ory.sh)

**Pull request merged**
